File: Ruby/p220.rb
# Author::	PJensen (mailto:jensen.petej@gmail.com)
# Copyright:: Copyright (c) 2010 Pete Jensen 
require 'project_euler'

# The problem space
module ProjectEuler::Problem220

	# An instance of the heighway dragon.
	class HeighwayDragon < ProjectEuler::Problem
		A, B = 'aRbFR', 'LFaLb'
		# Initializes the heigway-dragon problem space by setting
		# the string sequence and calling the super constructor.
		def initialize
			# Call superconstructor
			super 220, 'Heighway Dragon'
		
			# Define the inital state of the sequence.
			@sequence = 'Fa'	
		end

		def generate
			@sequence.replace!('a', A)
		end

		# Solve
		def solve
			generate
		end
	end

	a = HeighwayDragon.new()
	a.solve
	
end

File: Python/p220.py
import string

a = 'aRbFR'
b = 'LFaLb'

def D(n):
  if (n <= 0):
     return 'Fa'
  else:
     return D(n - 1).replace('a', a)
     return D(n - 1).replace('b', b)

print D(1)
s = D(1)

i = []
O = {0:'N', 90:'E', 180:'S', 270:'W'}

x = 0; y = 0; o = 0
d1 = []

print s
print 
print 'D(2) := FaRbFRRLFaLbFRa'

for c in s:
   print c,
   if (c == 'F'):
       d1 = []
       print ' => ' + O[o % 360]
   else:
      if (c == 'R'):
         o += 90
      if (c == 'L'):
         o -= 90

print i
